cfme/intelligence/optimization.py

```diff
@@ -0,0 +1,154 @@
+from cfme.intelligence.reports.saved import SavedReportsCollection
+from cfme.modeling.base import BaseCollection
+from cfme.modeling.base import BaseEntity
+from cfme.utils.appliance.implementations.ui import CFMENavigateStep
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.appliance.implementations.ui import navigator
+from cfme.utils.wait import wait_for
+from widgetastic_manageiq import SearchBox
+
+
+class OptimizationView(BaseLoggedInPage):
+    title = Text('//*[@id="main-content"]//h1')
+    table = Table(
+        '//*[@id="main_div"]//table', column_widgets={"Action": Button("contains", "Queue Report")}
+    )
+    refresh = Button(title="Refresh the list")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.logged_in_as_current_user
+            and self.title.text == "Optimization"
+            and self.table.is_displayed
+            and self.navigation.currently_selected == ["Overview", "Optimization"]
+            and self.breadcrumb.locations == ["Overview", "Optimization"]
+        )
+
+
+class OptimizationReportAllView(OptimizationView):
+    table = Table('//*[@id="main_div"]//table')
+
+    @property
+    def is_displayed(self):
+        menu_name = self.context["object"].menu_name
+        return (
+            self.logged_in_as_current_user
+            and self.title.text == menu_name
+            and self.navigation.currently_selected == ["Overview", "Optimization"]
+            and self.breadcrumb.locations == ["Overview", "Optimization", menu_name]
+        )
+
+
+class OptimizationSavedReportDetailsView(OptimizationView):
+    field = SelectorDropdown("id", "filterFieldTypeMenu")
+    search_text = SearchBox(locator='//input[contains(@placeholder, "search text")]')
+    table = Table('//*[@id="main_div"]//table')
+
+    @property
+    def is_displayed(self):
+        context_obj = self.context["object"]
+        return (
+            self.logged_in_as_current_user
+            and self.title.text == context_obj.name
+            and self.table.is_displayed
+            and self.navigation.currently_selected == ["Overview", "Optimization"]
+            and self.breadcrumb.locations
+            == ["Overview", "Optimization", context_obj.parent.parent.menu_name, context_obj.name]
+        )
+
+
+@attr.s
+class OptimizationSavedReport(BaseEntity):
+    name = attr.ib()
+    run_at = attr.ib()
+    username = attr.ib(default=None)
+
+
+@attr.s
+class OptimizationSavedReportsCollection(BaseCollection):
+    ENTITY = OptimizationSavedReport
+
+
+@attr.s
+class OptimizationReport(BaseEntity):
+    menu_name = attr.ib()
+    runs = attr.ib(default=None)
+
+    _collections = {
+        "optimization_saved_reports": OptimizationSavedReportsCollection,
+        "saved_reports": SavedReportsCollection,
+        "reports": ReportsCollection,
+    }
+
+    @property
+    def last_run_at(self):
+        view = navigate_to(self.parent, OptimizationView)
```

Comment:
Why is the view class being passed as the destination here? Is there some black magic in navmazing I missed out on?? :)

I think you want `navigate_to(self.parent, 'All')` here, which should result in an instance of `OptimizationView`.